src/app/core/auth/abstract.interceptor.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { throwError, Observable } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+
+/**
+ * Abstract HTTP Interceptor
+ */
+@Injectable()
+export abstract class AbstractHttpInterceptor implements HttpInterceptor {
+	protected constructor(public router: Router) {}
+
+	intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+		return next.handle(req).pipe(
+			catchError(err => {
+				if (!err.bypassAuthInterceptors) {
```

Comment:
Is this check still needed?

Reply:
Nope! Good catch. I've removed it, as now every implementation of this abstract interceptor should call the handleError method, which is where any bypassing will be handled via the headers.